src/boundary.jl

```diff
@@ -213,10 +213,22 @@ end
 
 function boundary(g::TransformedGeometry)
   b = boundary(parent(g))
-  isnothing(b) ? nothing : transform(g)(b)
+  t = transform(g)
+  if isnothing(b)
+    nothing
+  elseif b isa Geometry
+    TransformedGeometry(b, t)
+  elseif b isa Mesh
+    TransformedMesh(b, t)
+  end
```

Comment:
@eliascarv it turns out we needed a more general method to delay the discretization, which enforces the `TransformedGeometry` and `TransformedMesh` 👍🏽